specification/protocol/otlp.md

```diff
@@ -145,16 +148,47 @@ was not delivered.
 
 #### OTLP/gRPC Response
 
-The server may respond with either a success or an error to the requests.
+The response MUST be the appropriate serialized Protobuf message (see below for
+the specific message to use in the [Success](#success),
+[Partial Success](#partial-success) and [Failure](#failures) cases).
+
+##### Success
 
 The success response indicates telemetry data is successfully processed by the
 server. If the server receives an empty request (a request that does not carry
 any telemetry data) the server SHOULD respond with success.
 
-Success response is returned via
-[Export*ServiceResponse](https://github.com/open-telemetry/opentelemetry-proto)
-message (`ExportTraceServiceResponse` for traces, `ExportMetricsServiceResponse`
-for metrics, `ExportLogsServiceResponse` for logs).
+On success, the server response MUST be
+a Protobuf-encoded [Export*ServiceResponse](https://github.com/open-telemetry/opentelemetry-proto) message
+(`ExportTraceServiceResponse` for traces,
+`ExportMetricsServiceResponse` for metrics and
+`ExportLogsServiceResponse` for logs).
+
+The server SHOULD respond with success no sooner than after successfully
+decoding and validating the request.
+
+The server MUST leave the `partial_success` field unset
+in case of a successfull response.
+
+##### Partial Success
+
+If the processing of the request is partially successfull
+(i.e. when the server accepts only parts of the data and rejects the rest) the
+server MUST respond with the same message types as specified on the Success case.
```

Comment:
I'd just copy the part from above down here. Maybe its possible to merge with the paragraph below